MdeModulePkg/Bus/Pci/XhciDxe/XhciSched.c

```diff
@@ -1541,12 +1556,29 @@ XhciInsertAsyncIntTransfer (
   VOID  *Data;
   URB   *Urb;
 
-  Data = AllocateZeroPool (DataLen);
-  if (Data == NULL) {
-    DEBUG ((DEBUG_ERROR, "%a: failed to allocate buffer\n", __func__));
+  // MU_CHANGE [BEGIN]
+  EFI_STATUS  Status;
+
+  Status = Xhc->PciIo->AllocateBuffer (
+                         Xhc->PciIo,
+                         AllocateAnyPages,
+                         EfiBootServicesData,
+                         EFI_SIZE_TO_PAGES (DataLen),
+                         &Data,
+                         0
+                         );
+  if (EFI_ERROR (Status)) {

```

Comment:
```suggestion
  if (EFI_ERROR (Status) || Data == NULL) {
```

With setting Data = NULL before the allocate call. I think that will prevent CodeQL from being unhappy.

Reply:
done